bd-log-primitives/src/lib.rs

```diff
@@ -19,15 +19,52 @@ pub mod tiny_set;
 
 use crate::size::MemorySized;
 use ahash::AHashMap;
-pub use bd_proto::flatbuffers::buffer_log::bitdrift_public::fbs::logging::v_1::LogType;
+use bd_proto::protos::logging::payload::LogType;
+use protobuf::rt::WireType;
+use protobuf::{CodedInputStream, CodedOutputStream};
 use serde::{Deserialize, Serialize};
 use std::borrow::Cow;
 use std::sync::{Arc, LazyLock};
+use time::OffsetDateTime;
 
 pub const LOG_FIELD_NAME_TYPE: &str = "log_type";
 pub const LOG_FIELD_NAME_LEVEL: &str = "log_level";
 pub const LOG_FIELD_NAME_MESSAGE: &str = "_message";
 
+// Helpers for doing raw casts where we are sure the value fits and don't want to pay for
+// checks and avoid clippy lints.
+pub trait LossyIntToU32 {
+  fn to_u32(self) -> u32;
```

Comment:
should this be called to_u32_lossy just to make it obvious?

Reply:
Sure